compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/RootNodeOwner.skiko.kt

```diff
@@ -104,35 +107,36 @@ internal class RootNodeOwner(
     private val snapshotInvalidationTracker: SnapshotInvalidationTracker,
     private val inputHandler: ComposeSceneInputHandler,
 ) {
-    // TODO(https://youtrack.jetbrains.com/issue/COMPOSE-1257/Implement-new-FocusOwnerImpl-from-08.04.2024)
-    //  implement new changes from upstream
-    // TODO(https://github.com/JetBrains/compose-multiplatform/issues/2944)
-    //  Check if ComposePanel/SwingPanel focus interop work correctly with new features of
-    //  the focus system (it works with the old features like moveFocus/clearFocus)
     val focusOwner: FocusOwner = FocusOwnerImpl(
-        onRequestFocusForOwner = { focusDirection, _ ->
-            var result = platformContext.requestFocus()
-            if (result && focusDirection != null) {
-                result = platformContext.parentFocusManager.moveFocus(focusDirection)
-            }
-            return@FocusOwnerImpl result
+        onRequestFocusForOwner = { _, _ ->
+            platformContext.requestFocus()
         },
         onRequestApplyChangesListener = {
             owner.registerOnEndApplyChangesListener(it)
         },
-        onMoveFocusInterop = {
-            platformContext.parentFocusManager.moveFocus(it)
-        },
-        onFocusRectInterop = {
-            null
-        },
+        // onMoveFocusInterop's purpose is to move focus inside embed interop views.
+        // We use another logic inside SwingPanel
```

Comment:
It's skiko sourceset but it refers to desktop-only code (`SwingPanel`).

_Also I'd avoid "we" in comments where possible_

Reply:
Done